src/python/pants/pantsd/process_manager.py

```diff
@@ -290,61 +261,54 @@ def lifecycle_lock(self):
         )
 
     @property
-    def cmdline(self):
-        """The process commandline. e.g. ['/usr/bin/python2.7', 'pants.pex'].
-
-        :returns: The command line or else `None` if the underlying process has died.
-        """
-        with swallow_psutil_exceptions():
-            process = self._as_process()
-            if process:
-                return process.cmdline()
-        return None
+    def fingerprint(self):
+        """The fingerprint of the current process.
 
-    @property
-    def cmd(self):
-        """The first element of the process commandline e.g. '/usr/bin/python2.7'.
+        This reads the current fingerprint from the `ProcessManager` metadata.
 
-        :returns: The first element of the process command line or else `None` if the underlying
-                  process has died.
+        :returns: The fingerprint of the running process as read from ProcessManager metadata or `None`.
+        :rtype: string
         """
-        return (self.cmdline or [None])[0]
+        return self.read_metadata_by_name(self.name, self.FINGERPRINT_KEY)
 
     @property
     def pid(self):
         """The running processes pid (or None)."""
-        return self._pid or self.read_metadata_by_name(self._name, "pid", int)
+        return self.read_metadata_by_name(self._name, self.PID_KEY, int)
+
+    @property
+    def process_name(self):
```

Comment:
can add a type anno here too

Reply:
That's challenging due to the fact that the `read_metadata_by_name` method is generic on the function you pass to it. Will punt for now.